kuyruk/kuyruk.py

```diff
@@ -59,11 +59,20 @@ def wrapper(f: Callable) -> Task:
     @contextmanager
     def channel(self) -> Iterator[amqp.Channel]:
         """Returns a new channel from a new connection as a context manager."""
+        with self._connection_lock:
+            try:
+                ch = self._new_channel()
+            except amqp.exceptions.RecoverableConnectionError:
```

Comment:
Sometimes (e.g. for the test after `test_import_app_error()`), we can send heartbeat but cannot create channel and get `amqp.exceptions.ConnectionForced: (0, 0): (320) CONNECTION_FORCED - Closed via management plugin`. If we try to send heartbeat again, this time it fails. So, I think we can retry these once, which will open a new connection in the second run if necessary.